test/t8_geometry/t8_gtest_geometry_triangular_interpolation.cxx

```diff
@@ -0,0 +1,141 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element types in parallel.
+
+  Copyright (C) 2010 The University of Texas System
+  Written by Carsten Burstedde, Lucas C. Wilcox, and Tobin Isaac
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <gtest/gtest.h>
+#include <test/t8_gtest_macros.hxx>
+#include <t8_cmesh.hxx>
+#include <t8_cmesh/t8_cmesh_examples.h>
+#include <t8_geometry/t8_geometry_helpers.h>
+
+/* In this file we test the functionality of t8_geom_triangular_interpolation.
+ * The function maps reference coordinates inside the reference triangle or tetrahedron
+ * to the coordinate space of a triagle or tetrahedron given by its corner coordinates.
+ * We currently check that the reference coordinates of the corners are mapped correctly
+ * to the corners of the interpolated triangle/tet. */
+
+/* TODO:
+ *       - Check corner_value_dim > 1 
+ *       - Check different coefficient coordinates (for example midpoint)
+ *       - Check different corner values
+ *       - The 2d and 3d example contain a lot of nearly duplicate code.
+ *         Can we simplify?
+ */
+
+/* Check that the corner interpolation coordinates are 
+ * mapped to the corners of a triangle. */
+TEST (triangular_interpolation, corners_map_to_corners_2d)
+{
+  const double eps = 1e-14;
+
```

Comment:
We have macros for that
```suggestion
{
```

Reply:
i though so and looked for them but could not find them.

Where are they and where are they documented?

Reply:
Ah in t8.h.

I was looking in the gtest macros headers.